File: ulohy/23-krouzky/reseni/main.cs
using System;

internal class Program
{
    static void Main(string[] args)
    {
        int m = int.Parse(Console.ReadLine());
        Console.ReadLine();

        string[] mathClubMembers = new string[m];
        for (int i = 0; i < m; i++)
        {
            mathClubMembers[i] = Console.ReadLine();
        }

        Console.ReadLine();

        int b = int.Parse(Console.ReadLine());
        Console.ReadLine();

        string[] biologyClubMembers = new string[m];
        for (int i = 0; i < m; i++)
        {
            biologyClubMembers[i] = Console.ReadLine();
        }

        for (int i = 0; i < m; i++)
        {
            for (int j = 0; j < b; j++)
            {
                if (mathClubMembers[i] == biologyClubMembers[j])
                {
                    Console.WriteLine(mathClubMembers[i]);
                }
            }
        }
    }
}

File: ulohy/23-krouzky/reseni/main.cpp
#include <iostream>
#include <vector>
#include <string>

int main() {
    int m;
    std::cin >> m;

    std::vector<std::string> mathClubMembers(m);
    for (int i = 0; i < m; i++) {
        std::cin >> mathClubMembers[i];
    }

    int b;
    std::cin >> b;

    std::vector<std::string> biologyClubMembers(b);
    for (int i = 0; i < b; i++) {
        std::cin >> biologyClubMembers[i];
    }

    for (int i = 0; i < m; i++) {
        for (int j = 0; j < b; j++) {
            if (mathClubMembers[i] == biologyClubMembers[j]) {
                std::cout << mathClubMembers[i] << std::endl;
            }
        }
    }

    return 0;
}
